src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrState = [];
+  let newState = { ...state };
+
+  actions.forEach((action) => {
+    if (action.type === 'clear') {
+      newState = {};
+      arrState.push({ ...newState });
+    }
+
+    if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach((key) => delete newState[key]);
+      arrState.push({ ...newState });
+    }
+
+    if (action.type === 'addProperties') {
+      newState = {
+        ...newState,
+        ...action.extraData,
+      };
+      arrState.push({ ...newState });
+    }
+  });
```

Comment:
[CODE STYLE]: switch/case should always have a default case for error handling. After refactoring your if conditions into a switch statement, make sure to include a default case to handle any unexpected action types.